src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs

```diff
@@ -367,14 +365,18 @@ private IAntiforgeryFeature GetTokensInternal(HttpContext httpContext)
             return antiforgeryFeature;
         }
 
-        private void SetDoNotCacheHeaders(HttpContext httpContext)
+        /// <summary>
+        /// Sets the 'Cache-Control' header to 'no-cache, no-store' and 'Pragma' header to 'no-cache' overriding any user set value.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/>.</param>
+        protected virtual void SetDoNotCacheHeaders(HttpContext httpContext)
         {
             // Since antifogery token generation is not very obvious to the end users (ex: MVC's form tag generates them
             // by default), log a warning to let users know of the change in behavior to any cache headers they might
             // have set explicitly.
             LogCacheHeaderOverrideWarning(httpContext.Response);
 
-            httpContext.Response.Headers[HeaderNames.CacheControl] = "no-cache";
+            httpContext.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store";
```

Comment:
An alternative here would be to check if these directives already exist and append them if not present. This preserves other directives like `must-revalidate` and possible cache extensions. I'm satisfied with this approach since it looks like most clients will not cache entries with both `no-cache` and `no-store` but maybe someone should take a closer look.